lib/std/heap.zig

```diff
@@ -581,6 +581,31 @@ pub fn StackFallbackAllocator(comptime size: usize) type {
     };
 }
 
+var page_size: ?usize = null;
+
+pub fn pageSize() usize {
+    if (@as(?usize, switch (builtin.cpu.arch) {
+        .wasm32, .wasm64 => 64 * 1024,
+        else => null,
+    })) |pg_size| return pg_size;
```

Comment:
I don't see any reason why to expand the check. A better solution would be to have a fallback when no other options are available. Maybe we could have a `defaultPageSize` function or `default_page_size` field in `Target.Cpu`? Sparc seems to not use the typical 4k page size so that could be added into the Sparc target stuff in std. x86_64-linux can be executed on a kernel which doesn't have a 4k page size because you can configure that when building the kernel.

Reply:
> x86_64-linux can be executed on a kernel which doesn't have a 4k page size because you can configure that when building the kernel.

I'm not so sure about that. I checked the Linux kernel config spec, and it seems the only pagesize-related options (PAGE_SIZE_32KB, *K_PAGES etc) are architecture-specific (SuperH, Argonaut RISC, PA-RISC, Hexagon, PowerPC, Loongson, Aarch64, and MIPS). If we look at [the actual place the page size is set for x86 and x86_64](https://elixir.bootlin.com/linux/latest/source/arch/x86/include/asm/page_types.h), we see it's just hardcoded to `1<<12` (4096).

Moreover the [x86_64 manual](https://www.amd.com/content/dam/amd/en/documents/processor-tech-docs/programmer-references/24593.pdf) itself states [page 128]:

> The following physical-page sizes are supported: 4 Kbytes, 2 Mbytes, 4 Mbytes, and 1 Gbytes.

... and I don't think anybody would hack together a kernel with a default 2MiB page size other than as a joke 🙂.

Reply:
Even though it isn't usual to see it, still better to support it than to have bugs about it reported. If it's an option, then there's a chance someone somewhere is using it.

Reply:
For architectures where it's just rare (even very rare), then absolutely. For x86_64 though you'd have to extensively hack the kernel to make it work, if it's even theoretically possible in the first place; e.g. I tried manually increasing PAGE_SHIFT to 13 just now and it tripped a whole bunch of compiletime asserts.

IMHO I just don't think adding bytes and/or syscalls to almost every x86_64 Zig program is worth it just to support something that has never existed nor will realistically exist in the future.

Reply:
It just seems like a waste of time to add that code, plus I'm pretty sure glibc does perform a syscall for the page size regardless of architecture as well. This doesn't hurt performance as it only occurs once.